In [1]:
import numpy as np


# 1
aragonite - A form of calcium carbonate, less stable than calcite, pteropods make shells out of aragonite (~10% of ocean calcium carbonate)

calcite - A form of calcium carbonate, most stable form at all ocean temperatures and pressures, coccolithophores and foraminifera make shells out of calcite (~90% of ocean calcium carbonate)

calcium carbonate saturation horizon - The depth at which the ocean becomes saturated with respect to calcium carbonate. This occurs where $\Delta CO_3^{2-} = 0$ $(\Delta CO_3^{2-} = [CO_3^{2-}] - [CO_3^{2-}]_{sat})$

Rain ratio - The inorganic to organic carbon export ratio ($\frac{\phi_{CaCO_3}}{\phi_{C_{org}}}$)

# 2

In the 2-box model, the rate of change of the surface concentration of Carbon is defined by:

(1) $\frac{dC_s}{dt} = \frac{v}{V_s}(C_d - C_s) - \frac{\phi}{V_s}$

Where $C_s$ is the surface Carbon concentration, $C_d$ is the deep ocean Carbon concentration, $v$ is the surface-deep ocean exchange rate, $V_s$ is the volume of the upper box (surface ocean), and $\phi$ is the carbon export rate.  If we assume steady state, we get:
 
(2) $\frac{dC_s}{dt} = 0 = \frac{v}{V_s}(C_d - C_s) - \frac{\phi}{V_s}$

(3) $\phi = v(C_d - C_s)$

Next, we define potential Alkilinity: $P_{Alk} = Alk + [NO_3^-]$.  Variations in potential Alkilinity are governed only by the formation and dissolution of Calcium Carbonate, not by photosynthesis or remineralization. Using the vertical gradient of potential Alkilinity, we can define the downward export of $CaCO_3$:

(4) $\phi_{CaCO_3} = v*r_{CaCo_3:Alk}*([P_{Alk}]_d - [P_{Alk}]_s)$

Where $r_{CaCo_3:Alk}$ is the ratio of Calcium Carbonate to Alkilinity.  We can also define the downward export of organic carbon in terms of the vertical gradient of Nitrate and the ratio of Carbon to Nitrogen ($r_{C:N}$):

(5) $\phi_{C_{org}} = v*r_{C:N}*([NO_3^-]_d - [NO_3^-]_s)$

The rain ratio is the ratio of inorganic to organic Carbon export, or $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}}$.  Dividing equation (4) by (5) gives the rain ratio:

(6) $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}} = \frac{v*r_{CaCo_3:Alk}*([P_{Alk}]_d - [P_{Alk}]_s)}{v*r_{C:N}*([NO_3^-]_d - [NO_3^-]_s)}$

The ratio of Carbon to Nitrogen ($r_{C:N}$) is determined by Redfield's ratio.  Thus, $r_{C:N}$ = 106/16.  The ratio of $CaCO_3$ to Alkilinity ($r_{CaCo_3:Alk}$) is determined by the chemical expression: Alk = HCO3 + 2CO3  Thus, $r_{CaCo_3:Alk}$ = 2. Plugging these numbers into equation (6) and simplifying gives:

(7) $\frac{\phi_{CaCO_3}}{\phi_{C_{org}}} = \frac{([P_{Alk}]_d - [P_{Alk}]_s)}{13.25*([NO_3^-]_d - [NO_3^-]_s)}$

# 3

### a
Biological activity is the primary driver of the remineralization or organic matter.  The dissolution of $CaCO_3$ is different because biological activity plays no role.  The process that drives calcium carbonate dissolution is physics or kinetics.

### b
Maximum remineralization of organic matter is found in the upper ocean, above 100 meters.  The dissolution of calcium carbonate occurs deeper than remineralization

### c
As organic matter falls through the water column, approximately 99% gets remineralized.  In contrast, approximately 50% of calcium carbonate gets dissolve although this is much less certain.

# 4

The following equations are used to calculate the saturation states of Calcite and Aragonite:

$\Delta CO_3^{2-} = [CO_3^{2-}]_{obs} - [CO_3^{2-}]_{sat}$

$\Omega = \frac{[CO_3^{2-}]_{obs}}{[CO_3^{2-}]_{sat}}$

where $[CO_3^{2-}]_{obs} = Alk - DIC$

In [4]:
DIC = 2300 #Dissolved inorganic carbon (micro-mol/kg)
Alk = 2400 #Alkilinity (micro-mol/kg)
cal_sat = 124.2 #Saturation concentration of Calcite at depth = 6000dbar, T = 0C, S = 35
arg_sat = 183.7 #Saturation concentration of Aragonite at depth = 6000dbar, T = 0C, S = 35

CO3_obs = Alk - DIC #observed concentration of CO3
delCO3_cal = CO3_obs - cal_sat #Saturation state Delta CO3 for calcite
delCO3_arg = CO3_obs - arg_sat #Saturation state Delta CO3 for aragonite
ohm_cal = CO3_obs/cal_sat #Saturation state omega for calcite
ohm_arg = CO3_obs/arg_sat #Saturation state omega for aragonite

print(delCO3_cal, ohm_cal)
print(delCO3_arg, ohm_arg)

-24.200000000000003 0.8051529790660226
-83.69999999999999 0.5443658138268918


For calcite, $\Delta CO_3^{2-}$ = -24.2 micro-mol/kg and $\Omega$ = 0.805

For aragonite, $\Delta CO_3^{2-}$ = -83.7 micro-mol/kg and $\Omega$ = 0.544

Both calcite and aragonite are undersaturated.

# 5

The saturation horizon is the depth at which $\Delta CO_3^{2-} = 0$ 

$\Delta CO_3^{2-} = [CO_3^{2-}]_{obs} - [CO_3^{2-}]_{sat}$ 

Thus at the saturation horizon, $[CO_3^{2-}]_{obs} = [CO_3^{2-}]_{sat}$

The saturation state with depth is roughly the same for the Atlantic and North Pacific so any difference in saturation horizon must be due to a difference in $CO_3^{2-}$ concentration. 

$[CO_3^{2-}]_{obs} = Alk - DIC$

Alkilinity is roughly constant with depth in the ocean. However, DIC increases from the Atlantic ocean to the North Pacific following the overturning circulation.  The increase in DIC leads to a corresponding increase in $[CO_3^{2-}]_{obs}$ which shifts the curve of $[CO_3^{2-}]_{obs}$ with depth closer to the saturation curve, thus increasing the saturation horizon.

# 6

An increase in ocean $CO_2$ from the atmosphere will lead to an increase in DIC. According to the equation $[CO_3^{2-}]_{obs} = Alk - DIC$, an increase in DIC will correspondingly decrease $[CO_3^{2-}]$ and $\Delta CO_3^{2-}$, assuming that Alkilinity remains constant. A decrease in $\Delta CO_3^{2-}$ will lead to a higher saturation horizon.

A decrease in $[CO_3^{2-}]$ will decrease $\Omega$ meaning the ocean will become more undersaturated with respect to $[CO_3^{2-}]$.  Following the equation for the : $\frac{d[CaCO_3]}{dt} = - [CaCO_3]* k_{CaCO_3} * (1-\Omega)^n$ if $\Omega$ decreases, the rate of calcium carbonate dissolution will increase.




# 7

The following equation is used to solve 7a and b:

$\frac{d[CaCO_3]}{dt} = -w_{sink}\frac{[CaCO_3]}{dz} - [CaCO_3]k(1-\Omega (z))^n$

In [1]:
# ------- a ------- #
w_sink = 50 #sinking speed (m/day)
n = 1 #first order dissolution kinetics
k = 0.01 #day^-1
initial = 1 #initial fraction of CaCo3 (at 100 m)
distance = 4000-100 #sinking distance (m)
days = distance/w_sink #days to sink to 4000 m

omega = 0.9

rate = -k*(1-omega)**n #fraction lost per day

lost = days*rate #total fraction lost

final = initial + initial*lost #final fraction of CaCO3 that arrives at 4000m

print(f'The fraction of CaCO3 that arrives at 4000m for case a is {final*100}%')



The fraction of CaCO3 that arrives at 4000m for case a is 92.2%


### b
89% of calcium carbonate reaches 4000 meters

see image 7b in folder for work